README.md

```diff
@@ -1,2 +1,407 @@
+        WindowGroup {
+            if let user = userService.user {
+                // Returns a new instance of a `LoggedInContentView` that has a forwarded, non-optional User property.
+                loggedInContentViewInstantiator.instantiate(user)
+            } else {
+                // Returns a new instance of a `LoggedOutContentView`.
+                loggedOutContentViewInstantiator.instantiate()
+            }
+        }
+    }
+
+    public init(loggedInContentViewInstantiator: ForwardingInstantiator<User, LoggedOutContentView>, loggedOutContentViewInstantiator: Instantiator<LoggedOutContentView>, userService: UserService) {
+        self.loggedInContentViewInstantiator = loggedInContentViewInstantiator
+        self.loggedOutContentViewInstantiator = loggedOutContentViewInstantiator
+        self.userService = userService
+    }
+
+    /// A private property that knows how to instantiate a logged-in content view.
+    @Instantiated
+    private let loggedInContentViewInstantiator: ForwardingInstantiator<User, LoggedOutContentView>
+
+    /// A private property that knows how to instantiate a logged-out content view.
+    @Instantiated
+    private let loggedOutContentViewInstantiator: Instantiator<LoggedOutContentView>
+
+    /// An instance of an observable `UserService` that is instantiated when `MyApp` is instantiated. This object updates whenever the user changes.
+    @ObservedObject
+    @Instantiated
+    private var userService: UserService
+}
+```
+
+It is possible to write a `ForwardingInstantiator` with a type-erased second generic by utilizing `@Instantiated`‘s `fulfilledByType` parameter.
+
+### Creating the root of your dependency tree
+
+SafeDI automatically finds the root(s) of your dependency tree, and creates an extension on each root that contains a `public init()` function that instantiates the dependency tree.
+
+An `@Instantiable` type qualifies as the root of a dependency tree if and only if:
+
+1. The type‘s SafeDI-injected properties are all `@Instantiated`
+2. The type is not instantiated by another `@Instantiable` type
+
+### Comparing SafeDI and Manual Injection: Key Differences
+
+SafeDI is designed to be simple to adopt and minimize architectural changes required to get the benefits of a compile-time safe DI system. Despite this design goal, there are a few key differences between projects that utilize SafeDI and projects that don‘t. As the benefits of this system are clearly outlined in the [Features](#features) section above, this section outlines the pattern changes required to utilize a DI system like SafeDI.
+
+#### Instantiating objects
+
+In a manual DI system, it is common to directly call your dependencies‘ `init(…)` functions. When utilizing SafeDI, you must rely on `@Instantiated`-decorated properties to instantiate your dependencies for you. Calling a dependency‘s `init(…)` function directly effectively exits the SafeDI-built dependency tree, which removes property lifecycle guarantees.
```

Comment:
With the exception of the root, right?

Reply:
Good call that I should make this explicit. Yeah, for the root of the dependency graph you must call `init()`. It's still true that you shouldn't call `init(...)` though... but yeah I should make this more clear.

Reply:
521a99a4c308e9bceef1065372647cb75fe85266